ocaml/libs/sexpr/sExpr.ml

```diff
@@ -36,28 +38,31 @@ let unescape_buf buf s =
  * that have guaranteed invariants and optimised performances *)
 let escape s =
   let open Astring in
-  let escaped = Buffer.create (String.length s + 10) in
-  String.iter
-    (fun c ->
-      let c' =
+  if String.exists is_escape_char s then (
+    let escaped = Buffer.create (String.length s + 10) in
+    String.iter
+      (fun c ->
         match c with
         | '\\' ->
-            "\\\\"
+            Buffer.add_string escaped "\\\\"
         | '"' ->
-            "\\\""
+            Buffer.add_string escaped "\\\""
         | '\'' ->
-            "\\\'"
+            Buffer.add_string escaped "\\\'"
         | _ ->
-            Astring.String.of_char c
-      in
-      Buffer.add_string escaped c'
-    )
-    s ;
-  Buffer.contents escaped
+            Buffer.add_char escaped c
```

Comment:
Just a question: would a pipe operator be of the same effect (no temporary variable)?
E.g.
`(match c with ..) |> Buffer.add_string escaped`